### PR TITLE
Vulnerability section

### PR DESCRIPTION
**Pull request by zpevma**
9 of 14 elements implemented for Vulnerability section.

New example `cvrf_example_vulnerabilities.xml` can be used for testing.

### CODE REVIEW

File: cvrf2csaf/section_handlers/vulnerability.py [hunk 1]
```diff
@@ -1,13 +1,66 @@
...
+        if hasattr(root_element, 'CVE'):
+            # "^CVE-[0-9]{4}-[0-9]{4,}$" differs from the CVRF regex. Will be checked by json schema validation.
+            # TODO: double check, that no regex should be implemented here
+            vulnerability['cve'] = root_element.CVE.text
+
+        if hasattr(root_element, 'CWE'):
+            CWE_elements = root_element.CWE
+            if len(CWE_elements) > 1:
+                logging.warning(f'{len(CWE_elements)} CWE elements found, using only the first one.')
+            vulnerability['cwe'] = {}
```
Comment (sthagen) [inline cvrf2csaf/section_handlers/vulnerability.py:34]: Suggest to directly set the dict literal instead.

File: cvrf2csaf/section_handlers/vulnerability.py [hunk 2]
```diff
@@ -1,13 +1,66 @@
...
+            vulnerability['cwe']['id'] = CWE_elements[0].attrib['ID']
+            vulnerability['cwe']['name'] = CWE_elements[0].text
+
+        if hasattr(root_element, 'DiscoveryDate'):
+            vulnerability['discovery_date'] = root_element.DiscoveryDate.text
+
+        if hasattr(root_element, 'ID'):
+            vulnerability['id'] = {}
```
Comment (sthagen) [inline cvrf2csaf/section_handlers/vulnerability.py:42]: Same here (dict literal with assignments)

**Review by sthagen — APPROVED** [review]

File: cvrf2csaf/cvrf2csaf.py [hunk 3]
```diff
@@ -59,39 +59,29 @@ def __init__(self, config):
...
+            'DocumentType': self.document_category,
+            'DocumentDistribution': self.document_distribution,
+            'DocumentNotes': self.document_notes,
+            'DocumentPublisher': self.document_publisher,
+            'DocumentReferences': self.document_references,
+            'DocumentTitle': self.document_title,
+            'DocumentTracking': self.document_tracking,
+            'ProductTree': self.product_tree,
+            'Vulnerability': self.vulnerability,
+        }
```
Comment (cgi1) [inline cvrf2csaf/cvrf2csaf.py:75]: @tschmidtb51, @sthagen: Here is the direct assignment you wished for (no more if/elif statement here). :)

File: cvrf2csaf/section_handlers/vulnerability.py (hunk 1 above)
Comment (zpevma) [reply cvrf2csaf/section_handlers/vulnerability.py]: Good point, done.

**Comment by cgi1** [comment]
Talked through with @tschmidtb51, ready for Merge! @sustefil

**Pull request merged**